File: Repositories/electricgarden-eg-device-hq-9bbac4f2906a/api/funcs/shared/samples/sample.ts
export interface Sample {
  timestamp: Date;
  probeSoilTemp?: number;
  probeAirTemp?: number;
  probeMoisture?: number;
  ambientTemp?: number;
  ambientHumidity?: number;
  batteryVoltage?: number;
  rssi?: number;
  light?: number;
  snr?: number;
  errorCode?: number;
  co2?: number;
}

File: Repositories/electricgarden-kiosk-6f20b717c445/api/Models/Sample.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace KioskApi.Models
{
    [Table("sample")]
    public class Sample
    {
        public Sample(DeviceInstallation install, DateTimeOffset timestamp)
        {
            Installation = install;
            Timestamp = timestamp;
        }

        protected Sample()
        {
        }

        [Required] public int InstallationId { get; protected set; }

        public virtual DeviceInstallation Installation { get; protected set; }

        [Required] public DateTimeOffset Timestamp { get; protected set; }

        public double? ProbeSoilTemp { get; set; }

        public double? ProbeAirTemp { get; set; }

        public double? ProbeMoisture { get; set; }

        public double? AmbientHumidity { get; set; }

        public double? BatteryVoltage { get; set; }

        public double? Light { get; set; }

        public double? Rssi { get; set; }

        public double? Snr { get; set; }
        
        public double? Co2 { get; set; }

        public byte? ErrorCode { get; set; }
    }
}